### PR TITLE
#5 #13 Added support for votes on topics and styles topics card

### PR DESCRIPTION
**Pull request by maxletourneur**
- Added support for hasCurrentUserVoted on topic
- Added vote action for current user
- Added vote cancel action from current User

Current implementation is with the "Stack overflow like vote up/down" probably to change in the future.

### CODE REVIEW

**Comment by maxletourneur** [comment]
Issue #5

File: public/elements/vu-topic.html [hunk 1]
```diff
@@ -151,12 +176,60 @@
...
+
+        this.votes[this.id].splice(voteToCancelIndex,1);
+        this.notifyPath('votes.' + this.id);
+      },
+
+      setHasCurrentUserVoted: function(currentUserId, votes, currentSpace){
+        this.hasCurrentUserVoted = 0;
+
+        if (votes && votes[this.id]) {
+          let self = this;
```
Comment (mbuckley) [inline public/elements/vu-topic.html]: @maxletourneur remove self here and pass "this" to the map function. It's optional and will set the context of "this" so you don't have to use self = this...

File: public/elements/vu-topic.html [hunk 2]
```diff
@@ -151,12 +176,60 @@
...
+
+          votes[this.id].map(function(vote){
```
Comment (mbuckley) [inline public/elements/vu-topic.html]: @slapheadted and I are in your code :)

```
votes[this.id].map(function(vote){
  if(vote.createdBy === currentUserId){
    this.hasCurrentUserVoted = 1;
  }
}, this);
```

**Comment by kbwatts** [comment]
LGTM 👍

**Pull request merged**